include/envoy/network/listener.h

```diff
@@ -179,6 +192,31 @@ class Listener {
 
 typedef std::unique_ptr<Listener> ListenerPtr;
 
+/**
+ * A Udp listener interface.
+ */
+class UdpListener : public virtual Listener {
+public:
+  virtual ~UdpListener() {}
+
+  /**
+   * @return Event::Dispatcher& the dispatcher backing this listener.
+   */
+  virtual Event::Dispatcher& dispatcher() PURE;
+
+  /**
+   * @return the local address of the socket.
+   */
+  virtual const Network::Address::InstanceConstSharedPtr& localAddress() const PURE;
+
+  /**
+   * Send data through the underlying udp socket. If the send buffer of the socket FD is full, it is
+   * possible the write is dropped.
```

Comment:
Can you specify under what conditions the buffer is drained? Presumably always?

Reply:
yes